unittests/ProfilerTest.cpp

```diff
@@ -0,0 +1,41 @@
+#include "util/Profiler.h"
+#include <gtest/gtest.h>
+#include <thread>
+
+namespace util::detail {
```

Comment:
The tests should not be in a namespace themselves. Let's just `using namespace util;` etc. 
Take a look at other unit-tests we have for an example :+1: